loopy/transform/callable.py

```diff
@@ -365,7 +367,7 @@ def _inline_call_instruction(caller_knl, callee_knl, call_insn):
     rule_mapping_context = SubstitutionRuleMappingContext(callee_knl.substitutions,
                                                           vng)
     smap = KernelArgumentSubstitutor(rule_mapping_context, caller_knl,
-                                     callee_knl, arg_map)
+                                     callee_knl.copy(domains=new_domains), arg_map)
```

Comment:
Is this correct? Not entirely sure.

Reply:
It isn't. `new_domains`'s inames could be arbitrarily renamed, so they wouldn't make sense in the context of `callee_knl`. On top of that, who'd be looking at that domain, anyhow? The test here don't expose a necessity for this either.

I've reverted this, LMK if you disagree.

Reply:
I've added a test that fails with the revert of this change, but passes with this change.
`simplify_using_aff` which is called in `KernelArgumentSubstitutor` will be looking at that domain to simplify the indices.

Reply:
`test_auto_simplify_indices` should fail now

Reply:
OK, I've looked a bit more closely, and I now agree that it's (super inelegant and) OK. I've added some noisy warning comments to avoid accidents.